### ISSUE TITLE
DTS Bundling not exporting previously available interfaces and types

### ISSUE THREAD

**Issue by AndrewLeedham** [vanilla-extract-css/vanilla-extract#1115]
### Describe the bug

Since `@vanilla-extract/css@1.9.2` I cannot compile Vanilla Extract types with tsc. The issue seem to step from the bundling of dts files, as interfaces like `AllQueries` are no longer exported, so the TypeScript compiler throws the following error:
```
Exported variable 'exampleRecipe' has or is using name 'AllQueries' from external module "file:///node_modules/@vanilla-extract/css/dist/vanilla-extract-css.cjs" but cannot be named.
```
This happens when composing types like `StyleRule` which are exported but use interfaces that aren't.

### Reproduction

https://www.typescriptlang.org/play?ts=5.1.3#code/JYWwDg9gTgLgBAbzgZRgTwDYFMBKBXbOAXzgDMoIQ4ByAAQDcBDAO2Aw0YFosAPGKRgGMYAekEBncdQDcAKFmhIsRHChZBwMFmJkKVOk1bsuvfkNFqNWqXNmCIzcfFIRBecQC4U6bPmwBtanEsbGFoKQBdOABeRFk4GgAyDxc3cU56YHFgACNsai8EeIS4ewxoL2o8vCxqYqJZBtleJXh7R3heRnBfdU1tWMt+gAoihJzGYMLihLKKmjUAE2oAGnq1hKYoYBYYTziSuEWssA40acPx8sEAawvLhOPxU8ZzmjzXG9WZkqINy+owGYGCBWE4H1uBQODyOJzOlSBIOYYIhXx+CT+6LIrnc9wewVCMHCeJhADpyal3P8Hg0aT9MRjGgBKaRAA

### System Info

```shell
I cannot run that in the TS playground, but I am seeing the issue locally with:

  System:
    OS: macOS 13.3.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 267.59 MB / 16.00 GB
    Shell: 5.8 - /usr/local/bin/zsh
  Binaries:
    Node: 16.18.0 - ~/.nvm/versions/node/v16.18.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.6.3 - ~/.nvm/versions/node/v16.18.0/bin/npm
  Browsers:
    Safari: 16.4
  npmPackages:
    @vanilla-extract/esbuild-plugin: 2.2.2 => 2.2.2 
    @vanilla-extract/vite-plugin: 3.8.2 => 3.8.2 
    esbuild: 0.17.19 => 0.17.19 
    rollup: 2.79.1 => 2.79.1 
    vite: 3.2.7 => 3.2.7
```

### Used Package Manager

yarn

### Logs

```shell
Exported variable 'exampleRecipe' has or is using name 'AllQueries' from external module "file:///node_modules/@vanilla-extract/css/dist/vanilla-extract-css.cjs" but cannot be named.
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by mrm007** [comment]
Hi @AndrewLeedham, this looks like a quirk of the TypeScript compiler. We wouldn't want to expose APIs that we'd like to reserve the right to change in the future. However, you can work around this using the `satisfies` operator:

```ts
const focus = {
  '&:focus-visible': {
    color: 'blue'
  }
} satisfies StyleRule['selectors'];
```

[Does this work for you?][playground] I realise this is just an example and might not be the same as your production code.

[playground]: https://www.typescriptlang.org/play?ts=5.1.3#code/JYWwDg9gTgLgBAbzgZRgTwDYFMBKBXbOAXzgDMoIQ4ByAAQDcBDAO2Aw0YFosAPGKRgGMYAekEBncdQDcAKFmhIsRHChZBwMFmJkKVOk1bsuvfkNFqNWqXNmCIzcfFIRBecXAC8iWXBoAyAC4XN3FOemBxYAAjbGpAnz8-ewxoBOpYvCxqX2JZEnFGGEjSYCwPVExcAiwAbWpxLGxhaCkAXVteJXh7R3heRnBsHHVNbW9LMYAKBFzoxkaE2aS4FLSaNQATagAaXKI9vyYoYBYYcSXcv03IsA40S5W-WNcAa0en69v79JfBV92VySByBfmowGYGAhWE4fwBH0+N3Ed0YDxoEKhzBhcJynxBnxC7gRT0azRgrWJnwAdDTCeJDp88nigfi8kQAJTSIA

**Comment by AndrewLeedham** [comment]
@mrm007 thanks for the workaround. You are right this is a minimal repro. The actual use case is using helper functions which return those objects. I could probably use satisfies and infer return types, but this seems like quite a significant gotcha. Composing types at least in our codebase is very common, and we are not yet on TS 5 (although I will work on upgrading soon).

Perhaps I exported too much, I was just concious that other functions may have similiar issues. But I could just export `AllQueries` as its constituent parts are already exported?

**Comment by AndrewLeedham** [comment]
@mrm007 another case where keeping raw typing around doesn't quite work is conditionally modifying a styles object: [TS Playground Link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgZRgTwDYFMBKBXbOAXzgDMoIQ4ByAAQDcBDAO2Aw0YFosAPGKRgGMYAekEBncdQDcAKFm9IsMnmbDgEZnADmWZlgEwsAGWDMA1sizZh0cQApBEDNABcccfzPaANHAAWEPQGAMLO0AD87p5Q3gCUiLJwcE7Mnh7WWLZQ4nAAvInJydSM1O4ISUXJTi5QPpXJRJUk4owwwOKkwFi5qJi4BFgA2tTimdlSALpylcCkcI7hUAkVVWM2MHYjjK6BwVDUk-mFVSlL7nuhSw3EzfLJUFgweFBa61mbOXJEQA)

I updated the original PR to just export `AllQueries`.

**Comment by askoufis** [comment]
> @mrm007 another case where keeping raw typing around doesn't quite work is conditionally modifying a styles object: [TS Playground Link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgZRgTwDYFMBKBXbOAXzgDMoIQ4ByAAQDcBDAO2Aw0YFosAPGKRgGMYAekEBncdQDcAKFm9IsMnmbDgEZnADmWZlgEwsAGWDMA1sizZh0cQApBEDNABcccfzPaANHAAWEPQGAMLO0AD87p5Q3gCUiLJwcE7Mnh7WWLZQ4nAAvInJydSM1O4ISUXJTi5QPpXJRJUk4owwwOKkwFi5qJi4BFgA2tTimdlSALpylcCkcI7hUAkVVWM2MHYjjK6BwVDUk-mFVSlL7nuhSw3EzfLJUFgweFBa61mbOXJEQA)
> 
> I updated the original PR to just export `AllQueries`.

Not sure what the `a` is doing in your `selectors` object, but the `satisfies` isn't helping here:
```ts
const selectors = {
  'a': {
    color,
  }
// Keeps the type as narrow as possible
} satisfies StyleRule['selectors'];

if (color) {
  // selectors is not a StyleRule['selectors'], so it doesn't know about the `a:hover` property
  selectors['a:hover'] = {
    color: hoverColor
  }
}
```

By using an explicit type annotation, [you can conditionally mutate the `selectors` object](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3nAykgNigJQFd84BfOAMyghDgHIABANwEMA7YXXVgWhQAeMKKwDGMAPSiAztPoBuAFCLBkWFSLtxwCOzgBzFOxQiYKADLB2Aaywp846NIAUoiLmgAuONOFX9ADRwABYQzCYAwu7QAPzevlD+AJTeOIj4xPgA2vTS9iiOUHIAuhiKcHBu7L4++YXSqXiEJCg5eQ4wTvSlALwYZErlcMCUcM6h4VBRHlBJZRUV7QWdRTmeEybdcH3oQwuV0VDeG1OHe+RDZMoVUCgwRFB6S-VKZEA). You do unfortunately need to annotate the return type as well.

**Comment by AndrewLeedham** [comment]
@askoufis you are right I borked the syntax a little there, maybe something more [like this](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgZRgTwDYFMBKBXbOAXzgDMoIQ4ByAAQDcBDAO2Aw0YFosAPGKRgGMYAekEBncdQDcAKFm9IsMnmbDgEZnADmWZlgEwsAGWDMA1sizZh0cQApBEDNABcccfzPaANHAAWEPQGAMLO0AD87p5Q3gCUiLJwcE7Mnh7WWLZQ4nAAvInJydQAZNTuCElFyU4uUD5VyURVJOKMMMDipMBYuaiYuARYANrU4pnZUgC6clXApHD2gcFQYXUJldXjNjB2oyWuywbUU-mF1SnhUO5Hq1eNxC3yyVBYMHhQWttZuzlyREA). However, using an explicit type annotation is what caused the issue with recipes, so yes this would fix that example but re-introduce the issue from the first one.
